source/blink.cpp

```diff
@@ -308,6 +352,47 @@ void blink::application::read_import_address_table(const BYTE *image_base)
 	}
 }
 
+bool blink::application::set_watch(
+	const size_t dir_index,
+	std::vector<scoped_handle> &dir_handles,
+	std::vector<scoped_handle> &event_handles,
+	std::vector<notification_info> &notification_infos)
+{
+	dir_handles[dir_index].reset(CreateFileW(_source_dirs[dir_index].native().c_str(),
+		GENERIC_READ, FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE,
+		nullptr, OPEN_EXISTING, FILE_FLAG_BACKUP_SEMANTICS | FILE_FLAG_OVERLAPPED, nullptr));
+	if (dir_handles[dir_index] == INVALID_HANDLE_VALUE)
+	{
+		print("  Error: Could not open directory handle.");
+		return false;
+	}
+	const size_t buffer_size = 4096;
+	notification_infos[dir_index].p_info = std::shared_ptr<FILE_NOTIFY_INFORMATION>((FILE_NOTIFY_INFORMATION*)malloc(buffer_size), free_delete());
+	if (NULL == notification_infos[dir_index].p_info)
+	{
+		print("  Error: Could malloc p_info.");
+		return false;
+	}
+	notification_infos[dir_index].overlapped = { 0 };
+	ZeroMemory(&(notification_infos[dir_index].overlapped), sizeof(OVERLAPPED));
```

Comment:
`.overlapped` is zeroed two times in a row here.

Reply:
Thanks. Using member initializer now.